File: sdkwork-ai-sdk-csharp/Api/Batches2Api.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class Batches2Api
    {
        private readonly HttpClient _client;

        public Batches2Api(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Cancel batch
        /// </summary>
        public async Task<BatchResponse?> CancelBatchAsync(string batch_id)
        {
            return await _client.PostAsync<BatchResponse>(ApiPaths.AiPath($"/v1/batches/{batch_id}/cancel"), null);
        }

        /// <summary>
        /// List batches
        /// </summary>
        public async Task<BatchListResponse?> ListBatchesAsync(Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<BatchListResponse>(ApiPaths.AiPath("/v1/batches"), query);
        }

        /// <summary>
        /// Create batch
        /// </summary>
        public async Task<BatchResponse?> CreateBatchAsync(BatchCreateRequest body)
        {
            return await _client.PostAsync<BatchResponse>(ApiPaths.AiPath("/v1/batches"), body);
        }

        /// <summary>
        /// Get batch
        /// </summary>
        public async Task<BatchResponse?> GetBatchAsync(string batch_id)
        {
            return await _client.GetAsync<BatchResponse>(ApiPaths.AiPath($"/v1/batches/{batch_id}"));
        }
    }
}

File: sdkwork-ai-sdk-swift/Sources/API/Batches2Api.swift
import Foundation

public class Batches2Api {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Cancel batch
    public func cancelBatch(batch_id: String) async throws -> BatchResponse? {
        let response = try await client.post(ApiPaths.aiPath("/v1/batches/\(batch_id)/cancel"), body: nil)
        return response as? BatchResponse
    }

    /// List batches
    public func listBatches(params: [String: Any]? = nil) async throws -> BatchListResponse? {
        let response = try await client.get(ApiPaths.aiPath("/v1/batches"), params: params)
        return response as? BatchListResponse
    }

    /// Create batch
    public func createBatch(body: BatchCreateRequest) async throws -> BatchResponse? {
        let response = try await client.post(ApiPaths.aiPath("/v1/batches"), body: body)
        return response as? BatchResponse
    }

    /// Get batch
    public func getBatch(batch_id: String) async throws -> BatchResponse? {
        let response = try await client.get(ApiPaths.aiPath("/v1/batches/\(batch_id)"))
        return response as? BatchResponse
    }
}
